File: BOJ/1260.cpp
#include <stdio.h>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

int n, m, v;
int dvisited[1010], bvisited[1010];
vector <int> graph[1010];
queue <int> q;

void dfs(int cur) {
	dvisited[cur] = 1;
	printf("%d ", cur);
	for (int i = 0; i < graph[cur].size(); i++) {
		int next = graph[cur][i];
		if (dvisited[next] == 0) {
			dfs(next);
		}
	}
}

void bfs(int start) {
	q.push(start);
	bvisited[start] = 1;
	while (!q.empty()) {
		int f = q.front();
		q.pop();
		printf("%d ", f);
		for (int i = 0; i < graph[f].size(); i++) {
			int next = graph[f][i];
			if (bvisited[next] == 0) {
				bvisited[next] = 1;
				q.push(next);
			}
		}
	}
}


int main() {
	scanf("%d %d %d", &n, &m, &v);
	for (int i = 0; i < m; i++) {
		int a, b;
		scanf("%d %d", &a, &b);
		graph[a].push_back(b);
		graph[b].push_back(a);
	}
	for (int i = 1; i <= n; i++) {
		if (graph[i].size() > 1) {
			sort(graph[i].begin(), graph[i].end());
		}
	}
	dfs(v);
	printf("\n");
	bfs(v);


	return 0;
}

File: BOJ/1260/1260.py
import sys
from collections import deque

# sys.stdin = open('BOJ/1260/example.txt')

input = sys.stdin.readline

N, M, V = map(int, input().rstrip().split())
graph = {}
queue = deque()
visited = [0 for _ in range(N + 10)]

for i in range(1, N+1):
  graph[i] = []

for _ in range(M):
  a, b = map(int, input().rstrip().split())
  
  graph[a] = graph[a] + [b]
  graph[b] = graph[b] + [a]

for i in range(1, N+1):
  graph[i].sort()

def dfs(node):
  visited[node] = 1
  print(node, end=' ')

  if len(graph[node]) == 0:
    return 

  for _, next in enumerate(graph[node]):
    if visited[next] == 0:
      dfs(next)
  
def bfs(node):
  global bfsResult

  visited = [0 for _ in range(N + 10)]
  visited[node] = 1
  queue.append(node)
  print(node, end=' ')

  while len(queue) > 0:
    front = queue.popleft()
    
    for _, next in enumerate(graph[front]):
      if visited[next] == 0:
        visited[next] = 1
        queue.append(next)
        print(next, end=' ')

dfs(V)
print(end='\n')
bfs(V)
print(end='\n')